server/src/main/java/org/elasticsearch/search/searchafter/SearchAfterBuilder.java

```diff
@@ -243,7 +243,7 @@ public static SearchAfterBuilder fromXContent(XContentParser parser) throws IOEx
                             break;
 
                         default:
-                            throw new AssertionError("Unknown number type []" + parser.numberType());
+                            throw new IllegalArgumentException("Unknown number type: [" + parser.numberType() + "]");
```

Comment:
Small comment, maybe an error message like "[search_after] does not accept numbers of type [BIG_DECIMAL]" would be clearer.

Reply:
+1